napari/utils/key_bindings.py

```diff
@@ -317,14 +241,38 @@ def _get_user_keymap():
     return USER_KEYMAP
 
 
-def _bind_user_key(key, func=UNDEFINED, *, overwrite=False):
+def _bind_user_key(key: KeyBindingLike, func=UNDEFINED, *, overwrite=False):
     """Bind a key combination to the user keymap.
 
     See ``bind_key`` docs for details.
     """
     return bind_key(_get_user_keymap(), key, func, overwrite=overwrite)
 
 
+def _vispy2appmodel(event) -> KeyBinding:
+    key, modifiers = event.key.name, event.modifiers
+    if len(key) == 1 and key.isalpha():  # it's a letter
+        key = key.upper()
+        cond = lambda m: True  # noqa: E731
+    elif key in _VISPY_SPECIAL_KEYS:
+        # remove redundant information i.e. an output of 'Shift-Shift'
+        cond = lambda m: m != key  # noqa: E731
+    else:
+        # Shift is consumed to transform key
+
+        # bug found on OSX: Command will cause Shift to not
```

Comment:
can you add a few test cases to show this?

Reply:
this is deleted code that was re-added. i dont know a good way to test this and i dont think its worth my time since the code for handling this will change in the next PR i work on related to handling keypress events